SIAB/abacus/blscan.py

```
@@ -0,0 +1,392 @@
+        r.pop(idx_remove)
+        # refresh the condition
+        ediff = max(e) - min(e)
+        conditioned = ediff < cndt_thr or len(e) == 5
+
+    try:
+        popt, pcov = curve_fit(f=_morse, 
+                               xdata=r, 
+                               ydata=e,
+                               p0=[e[-1] - min(e), 1.0, 2.7, min(e)])
+    except RuntimeError:
+        raise ValueError('Fitting failed. This may due to the ill-defined PES caused by low-quality pseudopotential')
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)